vllm/engine/arg_utils.py

```diff
@@ -555,6 +563,23 @@ def from_cli_args(cls, args: argparse.Namespace):
         return engine_args
 
     def create_engine_config(self, ) -> EngineConfig:
+
+        # bitsandbytes quantization needs a specific model loader
+        # so we make sure the quant method and the load format are consistent
+        if (self.quantization == "bitsandbytes" or
+            self.qlora_adapter_name_or_path is not None) and \
+            self.load_format != "bitsandbytes":
+            raise ValueError(
+                "BitsAndBytes quantization only supports 'bitsandbytes' load "
+                f"format, but got {self.load_format}")
+
+        if (self.load_format == "bitsandbytes" or
+            self.qlora_adapter_name_or_path is not None) and \
+            self.quantization != "bitsandbytes":
+            raise ValueError(
+                "BitsAndBytes load format can only be used with bitsandbytes'"
+                f" quantization, but got {self.quantization}")
```

Comment:
nit: let's throw exception if `self.quantization != "bitsandbytes" and self.qlora_adapter_name_or_path is not None`

Reply:
The case of 'self.quantization != "bitsandbytes" and self.qlora_adapter_name_or_path is not None', actually is already covered in the code from line 576 to 579.

So I rephrased the error messages to mention the case that "qlora_adapter_name_or_path is not None" as well.